app/forms/openid_connect_authorize_form.rb

```diff
@@ -137,7 +141,7 @@ def success_redirect_uri
 
   def error_redirect_uri
     URIService.add_params(
-      redirect_uri,
+      redirect_uri_matches_sp_redirect_uri? ? redirect_uri : nil,
```

Comment:
this line feels pretty important since it is what makes this feature work. Thoughts on putting it into its own method for clarity / obviousness?